systemtest/src/test/java/io/strimzi/systemtest/specific/SpecificST.java

```diff
@@ -100,20 +110,151 @@ void testRackAware() {
+    void testRackAwareConnectWrongDeployment() throws Exception {
+        installClusterOperator(NAMESPACE, CO_OPERATION_TIMEOUT_SHORT);
+
+        String wrongRackKey = "wrong-key";
+        String rackKey = "rack-key";
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewRack()
+                            .withTopologyKey(rackKey)
+                        .endRack()
+                        .addToConfig("replica.selector.class", "org.apache.kafka.common.replica.RackAwareReplicaSelector")
+                    .endKafka()
+                .endSpec().done();
+
+        KafkaClientsResource.deployKafkaClients(false, KAFKA_CLIENTS_NAME).done();
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(KAFKA_CLIENTS_NAME).get(0).getMetadata().getName();
+
+        LOGGER.info("Deploy KafkaConnect with wrong rack-aware topology key: {}", wrongRackKey);
+        KafkaConnectResource.kafkaConnectWithoutWait(KafkaConnectResource.defaultKafkaConnect(CLUSTER_NAME, CLUSTER_NAME, 1)
+                .editSpec()
+                    .withNewRack()
+                        .withTopologyKey(wrongRackKey)
+                    .endRack()
+                .addToConfig("key.converter.schemas.enable", false)
+                .addToConfig("value.converter.schemas.enable", false)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .endSpec().build());
+
+        PodUtils.waitForPendingPod(CLUSTER_NAME + "-connect");
+        List<String> connectWrongPods = kubeClient().listPodNames(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND);
+        String connectWrongPodName = connectWrongPods.get(0);
+        LOGGER.info("Waiting for ClusterOperator to get timeout operation of incorrectly set up KafkaConnect");
+        TestUtils.waitFor("Wait for KafkaConnect 'Unschedulable' condition due to wrong rackKey used.",
+                Constants.GLOBAL_POLL_INTERVAL, CO_OPERATION_TIMEOUT_SHORT * 2, () -> {
+                List<Condition> conditions = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getConditions();
+                for (Condition condition : conditions) {
+                    if (condition.getReason().matches("TimeoutException") && condition.getType().matches("NotReady")) {
+                        return true;
+                    }
+                }
+                return false;
+            });
+
+        KafkaConnect kc = KafkaConnectResource.kafkaConnectClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get();
+        Condition kcCondition = kc.getStatus().getConditions().get(0);
+        assertThat("NotReady", is(kcCondition.getType()));
+        assertThat("TimeoutException", is(kcCondition.getReason()));
```

Comment:
Why do you assert something, which is verified a few lines above?

Reply:
Thanks, left-over from previous test case (work).